site/how-it-works.md

```diff
@@ -0,0 +1,86 @@
+---
+title: How Weave Flux Works
+menu_order: 70
+---
+
+This page describes the goals of flux, how it achieves them and 
+significant architectural decisions. It is intentionally high level 
+to prevent it from being out of date too quickly.
+
+# Goals
+
+The overall goal of Flux is to automate the deployment of services.
+A typical use case would be:
+
+1. A developer merges their tested code into a branch which creates an
+   artifact as a (Docker) container
+2. Flux detects the presence of a new container and deploys that to an
+   orchestrator (e.g. Kubernetes)
+3. Flux writes the new cluster configuration to version control
+4. Changes are made visible through chat integration and audit logging
+ 
+Hence, the goal is to automate away the need for a developer to 
+interact with an orchestrator (which is a common source of accidental
+failure) or with the systems that ensure that the orchestrator is in
+a working state.
+
+# Implementation Overview
+
+The following describes how Flux achieves the goals.
+
+## Monitoring For New Images
+
+Flux reads a list of running containers from Kubernetes. 
+For each image, it will reach out to the container registry to obtain
+the most recently released tag.
+
+Users are able to see whether the container is running the latest 
+image. Users are able to manually update to a specific or latest 
+version of an image.
+
+When services are in an "automated" mode, the service will 
+periodically check to see whether there are any new images. If there 
+are, they are deployed.
+
+In order to access private registries, credentials may be required.
+
+## Deployment of Images
+
+Flux will only deploy different images. It will not re-deploy images 
+with the same tag.
+ 
+Once a list of new images have been established, it will alter the 
+configuration of the cluster to deploy the new images.
+
+## Integration with Version Control
+
+Whenever Flux alters the configuration of a cluster it will write the
+changes back to version control.
+
+In other words, the state that Flux reports is the state of the 
+cluster. If you make changes to the version control, Flux will not 
+deploy that new configuration.
```

Comment:
In the same example: https://www.weave.works/docs/net/latest/introducing-weave/
lots of yous. So sticking to yous.